File: code/0866/866.回文素数.cpp
/*
 * @lc app=leetcode.cn id=866 lang=cpp
 * @lcpr version=21401
 *
 * [866] 回文素数
 */

// @lc code=start
class Solution {
public:
    bool check(int n) {
        if (n < 2) return false;
        for (int i = 2; i <= n / i; i++)
            if (n % i == 0) return false;
        
        int t = 0, x = n;
        while (x) {
            t = 10 * t + x % 10;
            x /= 10;
        }
        return t == n;
    }

    int primePalindrome(int n) {
        while (true) {
            if (check(n)) return n;
            n++;
            if (1e7 < n && n < 1e8) n = 1e8;
        }
    }
};
// @lc code=end



/*
// @lcpr case=start
// 
// @lcpr case=end

// @lcpr case=start
// 
// @lcpr case=end

// @lcpr case=start
// 
// @lcpr case=end

 */

File: code/866.回文素数.py
#
# @lc app=leetcode.cn id=866 lang=python3
# @lcpr version=21401
#
# [866] 回文素数
#

# @lc code=start
class Solution:
    def primePalindrome(self, n: int) -> int:
        def is_prime(n):
            return n > 1 and all(n % x for x in range(2, int(n ** 0.5) + 1))
        
        def reverse(x):
            res = 0
            while x:
                res = 10 * res + x % 10
                x //= 10
            
            return res
        
        while True:
            if n == reverse(n) and is_prime(n):
                return n
            n += 1
            if 10 ** 7 < n < 10 ** 8:
                n = 10 ** 8
# @lc code=end



#
# @lcpr case=start
# 
# @lcpr case=end

# @lcpr case=start
# 
# @lcpr case=end

# @lcpr case=start
# 
# @lcpr case=end

#


